lib/AST/Decl.cpp

```diff
@@ -1007,11 +1007,21 @@ SourceRange Decl::getSourceRangeIncludingAttrs() const {
 }
 
 bool Decl::isInMacroExpansionInContext() const {
-  auto *dc = getDeclContext();
-  auto parentFile = dc->getParentSourceFile();
   auto *mod = getModuleContext();
   auto *file = mod->getSourceFileContainingLocation(getStartLoc());
 
+  auto parentFile = [&]() {
+    // For accessors, the storage decl is the more accurate thing to check
+    // since the entire property/subscript could be macro-generated, in which
+    // case the accessor shouldn't be considered "added by macro expansion".
+    if (auto *accessor = dyn_cast<AccessorDecl>(this)) {
+      auto storageLoc = accessor->getStorage()->getLoc();
+      if (storageLoc.isValid())
+        return mod->getSourceFileContainingLocation(storageLoc);
+    }
```

Comment:
We could split this logic out into a separate method specific to AccessorDecl, but it seems to me that this is the behavior that clients would expect for `isInMacroExpansionInContext`, even if it's not strictly based on the DeclContext

Reply:
Yeah I agree that what you have here is the intended behavior of `isInMacroExpansionInContext`